packages/next/src/cli/next-dev.ts

```diff
@@ -211,10 +211,7 @@ const nextDev: CliCommand = async (argv) => {
   // If neither --port nor PORT were specified, it's okay to retry new ports.
   const allowRetry =
     args['--port'] === undefined && process.env.PORT === undefined
-
-  // We do not set a default host value here to prevent breaking
-  // some set-ups that rely on listening on other interfaces
-  const host = args['--hostname']
+  const host = args['--hostname'] || '0.0.0.0'
```

Comment:
Seems this conflicts with https://github.com/vercel/next.js/pull/27306

Reply:
@ijjk Hmm can we also account for windows in `next dev`? Either way, it seems `next start` is limiting what we can change here if we don't make a change to it.